stablehlo/transforms/StablehloRefineShapes.cpp

```diff
@@ -585,6 +585,30 @@ struct RefineConvolutionOpPattern : public OpRewritePattern<ConvolutionOp> {
   }
 };
 
+struct RefineCustomCallOpPattern : public OpRewritePattern<CustomCallOp> {
+  using OpRewritePattern::OpRewritePattern;
+  LogicalResult matchAndRewrite(CustomCallOp op,
+                                PatternRewriter& rewriter) const override {
+    auto operandIndicesAttr = op->getAttr("indices_of_shape_operands")
+                                  .dyn_cast_or_null<DenseIntElementsAttr>();
+    if (!operandIndicesAttr)
+      return rewriter.notifyMatchFailure(
+          op, "expected an indices_of_shape_operands attribute");
+
+    SmallVector<ShapedTypeComponents> refinements;
+    for (auto operandIndexElt : operandIndicesAttr.getValues<APInt>()) {
+      int64_t operandIndex = operandIndexElt.getSExtValue();
+      if (operandIndex < 0 || operandIndex >= op->getNumOperands())
+        return rewriter.notifyMatchFailure(op, "expected valid operand index");
+      SmallVector<int64_t> refinement;
+      if (failed(matchInts(op->getOperand(operandIndex), refinement)))
+        return rewriter.notifyMatchFailure(op, "expected constant operand");
+      refinements.emplace_back(refinement);
```

Comment:
Do we need any validation for "all values must be positive" ?

Reply:
Good catch! I've updated the code here and opened a similar ticket for verification in general (#1170).

Reply:
lgtm